runtime/src/bank.rs

```diff
@@ -17410,9 +17411,9 @@ pub(crate) mod tests {
             recent_blockhash,
         );
         let result = bank.process_transaction(&tx);
-        assert_ne!(
```

Comment:
Because keeping the non-equality check should cover both the  `InvalidRentPayingAccount` error and the `InsufficientFundsForRent` error and the latter needs a "matches" non-equality check due to the index inside. I thought it was more straightforward to assert what the error actually is rather than what it's not, because that's implicit.